File: src/easy/FindLucky.swift
func findLucky(_ arr: [Int])-> Int {
	var nums = Array(repeating: 0, count: (arr.max() ?? 0) + 1)
	var index = nums.count - 1
	for num in arr {
		nums[num] += 1
	}
	while index > 0 {
		if nums[index] == index {
			return index
		}
		index -= 1
	}
	return -1
}

File: src/easy/find-lucky.ts
export const findLucky = (arr: number[]): number => {
    const nums = Array(Math.max(...arr) + 1).fill(0);
    for (const num of arr) {
        nums[num]++;
    }
    for (let i = arr.length; i > 0; i--) {
        if (i === nums[i]) return i;
    }
    return -1;
};
